### ISSUE TITLE
[US-14-5] Extension by another Page Object

### ISSUE THREAD

**Issue by savinovava** [jdi-testing/jdn-ai#246]
Need for tech decomposition
https://jdi-family.atlassian.net/wiki/spaces/JDN/pages/22347810/US-14-5+Extension+by+another+Page+Object

**Comment by SemyonKuznetsov** [comment]
[update old link](https://www.figma.com/file/DxLzXoAT23Gcu9DmCIuTcW/Design?node-id=1928%3A1527998)

**Comment by MariiaNebesnova** [comment]
should we allow to download extended PO without PO by that extended?

**Comment by MariiaNebesnova** [comment]
should we revise locators somehow?

**Comment by AlexeyGirin** [comment]
> should we revise locators somehow?
в дочерней страничке - удаляем все элементы (с одинаковыми локаторами+именем) которые есть в базовой страничке (что бы они не дублировались)

**Comment by AlexeyGirin** [comment]
> can the PO be empty?

Да, может. Сейчас базовую страничку любят делать пустой про запас

**Comment by AlexeyGirin** [comment]
Создать задачу на Семёна - что бы показывать содержимое базовой странички ввиде тултипа при наведении на название базового пэйдж обжекта

**Comment by SemyonKuznetsov** [comment]
наследование может быть почти бесконечным, но на практике показывают только последние

**Comment by SemyonKuznetsov** [comment]
делаем без локаторов родительского класса

**Comment by SemyonKuznetsov** [comment]
[figma link](https://www.figma.com/file/DxLzXoAT23Gcu9DmCIuTcW/Design?node-id=6509%3A335653)
только про чекбокс, основной редизайн страницы не закончен
<img width="611" alt="Снимок экрана 2022-09-28 в 18 01 18" src="https://user-images.githubusercontent.com/101407186/192814058-79355973-efba-46e9-b564-e3dbb6748695.png">


[figma link](https://www.figma.com/file/DxLzXoAT23Gcu9DmCIuTcW/Design?node-id=6504%3A335154)

<img width="444" alt="Снимок экрана 2022-09-28 в 17 58 31" src="https://user-images.githubusercontent.com/101407186/192813294-725b03fc-a9ce-4257-b2a1-676bcbb4fa38.png">

базовый РО фиксируется первым в списке при создании или после нажатия на "base class"
![#246 Extension PO 1](https://user-images.githubusercontent.com/101407186/192812943-9aceac07-4f0f-4eea-ab93-ff6c8759d69f.png)
![#246 Extension PO 2](https://user-images.githubusercontent.com/101407186/192812952-f0f39c93-7212-41bd-84a9-c4754a930fba.png)
![#246 Extension PO 3](https://user-images.githubusercontent.com/101407186/192812963-05763bfe-885e-4eb0-965b-f881a12551d2.png)
![#246 Extension PO 4](https://user-images.githubusercontent.com/101407186/192812967-0e47a3ee-b031-4af6-b649-74f5ecee454d.png)

**Comment by SemyonKuznetsov** [comment]
на дизайн

- доп меню для выбора баз. класса для расширения
- базовый класс - отменяемая штука, надо добавить такую возможность

**Comment by AlexeyGirin** [comment]
Делаем дерево.

**Comment by SemyonKuznetsov** [comment]
![image](https://user-images.githubusercontent.com/101407186/225870104-c2d17806-b699-426a-8707-a779b8c6e431.png)

**Comment by SemyonKuznetsov** [comment]
[Figma link](https://www.figma.com/file/DxLzXoAT23Gcu9DmCIuTcW/Design?node-id=8211-373018&t=vvO0J7SNnWMOwGY7-4)

Делаем базовый РО, после нажатия меню закрывается, сменяется иконка
![image](https://user-images.githubusercontent.com/101407186/225884231-a26a1b21-1073-4481-81d7-cd0acdb41f98.png)

Открываем меню, строчку selected. Можно вернуть пр повторном нажатии
![image](https://user-images.githubusercontent.com/101407186/225884334-bf9a7b52-bca7-47fc-8168-a878b464504e.png)

При условии наличия базового класса у всех РО, в том числе и у базовых, появляется в контекстном меню вариант "Extend"
Вопрос: стоит ли показывать эту строку всегда, просто в начале задизейблена будет?
![image](https://user-images.githubusercontent.com/101407186/225884713-45dadf77-3d75-4418-88cc-fcff9a8cd18e.png)

При наведении на который выводим доп меню с вариантами существующих сейчас базовых РО (минимум 1)
![image](https://user-images.githubusercontent.com/101407186/225885008-a21923d1-c019-44e2-a903-afa9c3f3fdf9.png)

После выбора чем экстендить у нас меняется иерархия
![image](https://user-images.githubusercontent.com/101407186/225885151-d1682680-b13d-45bb-81fe-f1f1d202cf26.png)

**Comment by savinovava** [comment]
Вопрос: стоит ли ограничить число РО в этой цепочке, ну например шестью, больше же никто не будет делать?
- pattern: не более 2ух вложенностей
